### DISCUSSION TITLE
How to back out of prisma.$transaction

### DISCUSSION THREAD

**Discussion by kokokenada** [prisma/prisma#3460] [Q&A] [answered]
If you are accumulating an array of promises to be dropped into prisma.$transaction, and part way through, you decide to abort, what are you supposed to do to ensure the created promises are not actually executed and you don't create a resource leak?

The docs https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/transactions describe sending a batch, but not how to back out.

**Comment by ryands17** [comment, marked as answer]
Hey @kokokenada 👋 
If you are accumulating promises and have not dropped them yet into `prisma.$transaction`, then not adding them to `$prisma.transaction` would be the way to go as those promises are not executed until `prisma.$transaction` is invoked. You can read more about this [here](https://github.com/prisma/docs/issues/800) and we will be adding this to the docs as well.

If you already have added them to `prisma.$transaction`, then currently there's no way to abort it. It would be great if you could add a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for that.

**Reply by kokokenada** [reply]
Thanks to the pointer re lazy evaluation.  That's what I needed to know.  Re the feature request, for the (overly) simplistic $transaction API, I think that a simple tweak to the docs describing your answer would be fine.  The real fix is to implement a better transaction API as discussed in https://github.com/prisma/prisma/issues/1844

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
